fossunited/api/profile.py

```diff
@@ -56,63 +56,89 @@ def toggle_profile_privacy(value):
 
 @frappe.whitelist()
 def update_profile(fields_dict):
+    """
+    Updates User Profile data.
+    Combined with Full Name and Username updates in User doctype,
+    whenever these fields are updated.
+    """
     user_doc = get_session_user_profile()
-
-    if not is_unique_username(
-        fields_dict.get("username"), user_doc.name
-    ):
-        frappe.throw("Username already exists")
-
     try:
+        updated_fields = {
+            "full_name": fields_dict.get("full_name"),
+            "username": fields_dict.get("username"),
+            "route": fields_dict.get("username"),
+            "bio": fields_dict.get("bio"),
+            "current_city": fields_dict.get("current_city"),
+            "about": fields_dict.get("about"),
+            "website": fields_dict.get("website"),
+            "x": fields_dict.get("x"),
+            "linkedin": fields_dict.get("linkedin"),
+            "github": fields_dict.get("github"),
+            "gitlab": fields_dict.get("gitlab"),
+            "instagram": fields_dict.get("instagram"),
+            "youtube": fields_dict.get("youtube"),
+            "devto": fields_dict.get("devto"),
+            "medium": fields_dict.get("medium"),
+            "mastodon": fields_dict.get("mastodon"),
+        }
+
         frappe.db.set_value(
-            USER_PROFILE,
-            user_doc.name,
-            {
-                "full_name": fields_dict.get("full_name"),
-                "username": fields_dict.get("username"),
-                "route": fields_dict.get("username"),
-                "bio": fields_dict.get("bio"),
-                "current_city": fields_dict.get("current_city"),
-                "about": fields_dict.get("about"),
-                "website": fields_dict.get("website"),
-                "x": fields_dict.get("x"),
-                "linkedin": fields_dict.get("linkedin"),
-                "github": fields_dict.get("github"),
-                "gitlab": fields_dict.get("gitlab"),
-                "instagram": fields_dict.get("instagram"),
-                "youtube": fields_dict.get("youtube"),
-                "devto": fields_dict.get("devto"),
-                "medium": fields_dict.get("medium"),
-                "mastodon": fields_dict.get("mastodon"),
-            },
+            USER_PROFILE, user_doc.name, updated_fields
```

Comment:
Should we try this with `ignore_permissions`? That way it will show on desk what changes were made by who.

Reply:
We could add it up. In fact I have tested with it too.

Ideally it's the User itself who should access the pages. And permission errors wouldn't come up here. Anyone other than the user (sans administrator) shouldn't change this as well.

Reply:
Not sure about that. Please test it with a non system user User. `ignore_permissions` should be gtg, but please test that once too.

Reply:
The profile updates work without the flag. New sign-up from UI itself cant be tested from local. But I created several users to see and signed up from their accounts to check the workflow.

Reply:
In this demo, Jorge is a non-system user created and he is updating is profile:

![user-test](https://github.com/user-attachments/assets/06c3cbf9-5c18-4d94-a8e7-4f78793faa99)

Reply:
@harshtandiya anything to add-on here ?